Dropped dependency on gutil and resolve write callback being called multiple times.

KenEucker opened the pull request:
This closes #1 and #2 

Also, I apologize for the many changed lines due to formatting. I can create another revision that only includes the changes for this update if need be.

tomcardoso commented:
I'll admit, I haven't used this gulp plugin in a few years. I'll make a point of running a quick test tomorrow, then I'll merge this all up. Thanks for this!

KenEucker commented:
Friendly bump.

tomcardoso commented:
Sorry this took so long. Just tested it out, and I'm running into an error ("Did you forget to signal async completion?"). Here's a log, along with a test repo… any thoughts?

```
➜  gulp-dropbox-test gulp
[01:10:15] Using gulpfile ~/Dropbox/code/Personal/gulp-dropbox-test/gulpfile.js
[01:10:15] Starting 'default'...
(node:47731) UnhandledPromiseRejectionWarning
(node:47731) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:47731) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
[01:10:16] The following tasks did not complete: default
[01:10:16] Did you forget to signal async completion?
```

I'm using node v8.11 here.

Test repo (I've pointed the `gulp-dropbox` install at your fork): https://github.com/tomcardoso/gulp-dropbox-test

KenEucker commented:
In testing, I'd found that the path being set for the upload was malformed and the error response from this plugin was not being handled. I've added code that now emits an error when the plugin fails as well as a call to path.resolve for ensuring that file paths are well formed.

This addresses the issues you found in the test. I've confirmed that your test repository works with the updated master branch of my fork with gulp version 3.9.1 and gulp ^4.0.0.

tomcardoso commented:
<img width="566" alt="screen shot 2019-02-21 at 12 48 08 am" src="https://user-images.githubusercontent.com/2408118/53146539-9abafb80-3572-11e9-9da9-64178468df44.png">
Awesome. 👍
